### PR TITLE
Fix Python deprecation warnings

### PR DESCRIPTION
**Pull request by greghudson**
Address all warnings issued by "python -Werror::DeprecationWarning"
in the test suite and doc build, as of Python 3.8.2.

### CODE REVIEW

**Comment by greghudson** [comment]
The Python 3.9 release notes recommended looking for deprecation warnings.  I found a bunch of string escaping nits and the thorny issue of importing runenv.py from k5test.py.

Importing a module from a specified source file has gone through several iterations in Python[1].  The only method not deprecated in current versions of Python uses a feature new in Python 3.5 (the test suite currently requires Python 3.2.4).  It's also three kind of inscrutable lines of code instead of a simple function call.

The approach taken in this PR is to add the top of the build directory to sys.path so that a simple `import runenv` will work.  I'm not really fond of that, but means we aren't fighting the language as much.

As an alternative, I prototyped changing Makefile.in to write out runenv in JSON and then changing k5test.py to load it with `json.load()'.  That produces a dictionary rather than a module, so everything that references `runenv.foo` has to change to referencing `runenv['foo']`.  (There are ways to convert a dictionary into an object, but none of them seem very natural.)  That's not so bad, but also, writing out JSON from bash-in-Makefile is pretty grotty--in particular, turning RUN_VARS into a comma-separated list of mappings (without writing out a comma at the end) isn't very elegant.

A third alternative might be to just execute the contents of runenv.py using `exec()`.  This would discard the namespacing (`runenv.env` would become just `env`), so perhaps some changes to the contents would make sense.  I haven't prototyped this option.

[1] https://stackoverflow.com/questions/19009932/import-arbitrary-python-source-file-python-3-3

**Review by frozencemetery — CHANGES_REQUESTED** [review]
I think this (i.e., messing with sys.path) is also my preference for the possible ways to handle the import.

File: doc/tools/doxybuilder_types.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def _process_typedef_node(self, node):
         # remove  macros
         t_definition = re.sub('KRB5_CALLCONV_C', '', t_definition)
         t_definition = re.sub('KRB5_CALLCONV', '', t_definition)
-        t_definition = re.sub('\*', '\\*', t_definition)
+        t_definition = re.sub(r'\*', '\\*', t_definition)
```
[the remark this answers is not in the document]
Comment (greghudson) [reply doc/tools/doxybuilder_types.py:114]: I had to instrument this and look at the resulting output to make sure, but it's not a no-op.  For the match expression, we want the regexp processor to see `\*` so it matches a literal asterix.  The substitution string is not a regexp, and we want it to also see `\*` so it substitutes in a backslash and an asterix.  `r'\\*'` would add two backslashes, which is not the current behavior.

[the remark this answers is not in the document]
Comment (frozencemetery) [reply doc/tools/doxybuilder_types.py:114]: I see, thanks for clarifying!

**Review by frozencemetery — APPROVED** [review]

**Pull request merged**